### PR TITLE
Add pachctl commands for all Auth GRPC methods

### PR DESCRIPTION
**Pull request by msteffen**

### CODE REVIEW

File: src/client/auth/auth.proto [hunk 1]
```diff
@@ -77,12 +77,9 @@ message ActivateResponse {}
 
 service API {
     rpc Activate(ActivateRequest) returns (ActivateResponse) {}
-
```
Comment (derekchiang) [inline src/client/auth/auth.proto]: Could you keep the newlines?  They are to distinguish different logical "sections" of the API.

File: src/client/client.go [hunk 2]
```diff
@@ -245,23 +248,43 @@ func (c *APIClient) connect() error {
...
-	c.cancel = cancel
 	return nil
 }
 
-func (c *APIClient) addMetadata(ctx context.Context) context.Context {
+// RequestCtx is a convenience function that returns a sensible default
+// context for top-level Pachyderm RPCs sent by the user (though it does not
+// assume a deadline, in case it's used by the command-line client.
+// Note that the client may cancel in-flight requests using this context if
+// it believes the server has become unavailable.
```
Comment (derekchiang) [inline src/client/client.go]: Do these changes still make sense given that we are using the Auth API directly?

File: src/server/auth/cmds/cmds.go [hunk 3]
```diff
@@ -21,6 +17,34 @@ import (
 
 var githubAuthLink = `https://github.com/login/oauth/authorize?client_id=d3481e92b4f09ea74ff8&redirect_uri=https%3A%2F%2Fpachyderm.io%2Flogin-hook%2Fdisplay-token.html`
 
+// ActivateCmd returns a cobra.Command to activate the security features of
+// Pachyderm within a Pachyderm cluster. All repos will go from
+// publicly-accessible to accessible only by the owner, who can subsequently add
+// users
```
Comment (derekchiang) [inline src/server/auth/cmds/cmds.go]: Activate should take `--admins` too.

File: src/server/auth/cmds/cmds.go [hunk 4]
```diff
@@ -79,8 +94,141 @@ func LoginCmd() *cobra.Command {
 	return login
 }
 
+// CheckCmd returns a cobra command that sends an "Authorize" RPC to Pachd, to
+// determine whether the specified user has access to the specified repo.
```
Comment (derekchiang) [inline src/server/auth/cmds/cmds.go]: What's the use case for this command?

File: src/server/auth/cmds/cmds.go [hunk 5]
```diff
@@ -79,8 +94,141 @@ func LoginCmd() *cobra.Command {
...
+			fmt.Printf("%t\n", resp.Authorized)
+			return nil
+		}),
+	}
+	return check
+}
+
+// GetCmd returns a cobra command that gets either the ACL for a Pachyderm
+// repo or another user's scope of access to that repo
+func GetCmd() *cobra.Command {
```
Comment (derekchiang) [inline src/server/auth/cmds/cmds.go]: Is this command going to look like `pachctl get`?  That sounds a little misleading.  Maybe `get-acl`?

File: src/server/auth/cmds/cmds.go (hunk 4 above)
Comment (msteffen) [inline src/server/auth/cmds/cmds.go]: My plan was that only the owner of a repo could get the ACL for that repo, but any user could check if they had access to it

File: src/client/auth/auth.proto (hunk 1 above)
Comment (msteffen) [inline src/client/auth/auth.proto]: Sure, done

File: src/client/client.go (hunk 2 above)
Comment (msteffen) [inline src/client/client.go]: Nope, this has been removed
Also, per discussion, https://github.com/pachyderm/pachyderm/pull/2070 means that there is no persistent client context anymore.

File: src/server/auth/cmds/cmds.go (hunk 3 above)
Comment (msteffen) [inline src/server/auth/cmds/cmds.go]: Done

File: src/server/auth/cmds/cmds.go [hunk 6]
```diff
@@ -21,6 +17,44 @@ import (
...
+			}
+			c, err := client.NewOnUserMachine(true, "user")
+			if err != nil {
+				return fmt.Errorf("could not connect: %s", err.Error())
+			}
+			_, err = c.AuthAPIClient.Activate(
+				c.Ctx(),
+				&auth.ActivateRequest{
+					ActivationCode: activationCode,
+				})
```
Comment (derekchiang) [inline src/server/auth/cmds/cmds.go]: You don't seem to be actually using `admins`.

Comment (msteffen) [inline src/server/auth/cmds/cmds.go]: Whoops. Fixed (and tested manually)

**Pull request merged**